### PR TITLE
Provide System.IO.Pipelines package readme

### PR DESCRIPTION
**Pull request by eNeRGy164**
Fill out details for the `System.IO.Pipelines` NuGet package readme.

See #92228

### CODE REVIEW

File: src/libraries/System.IO.Pipelines/src/PACKAGE.md [hunk 1]
```diff
@@ -2,43 +2,113 @@
...
+{
+    for (int i = 0; i < input.Length; i++)
+    {
+        Memory<byte> memory = writer.GetMemory(1);
+        memory.Span[0] = input[i];
+        writer.Advance(1);
+
+        // Make the data available to the PipeReader.
+        await writer.FlushAsync();
+    }
```
Comment (davidfowl) [inline src/libraries/System.IO.Pipelines/src/PACKAGE.md:53]: This example looks really inefficient.

Comment (eNeRGy164) [reply src/libraries/System.IO.Pipelines/src/PACKAGE.md:53]: I tried to create a simple implementation of the workings of `Pipe` but have no further experience with it in practice.

The example that is part of the _conceptual documentation_ link (<https://learn.microsoft.com/en-us/dotnet/standard/io/pipelines#pipe-basic-usage>) requires quite a large explanation, which didn't seem appropriate for the readme of the package itself.

I'm open for suggestions.

Comment (ViktorHofer) [reply src/libraries/System.IO.Pipelines/src/PACKAGE.md:53]: cc @area-system-io

Comment (ViktorHofer) [reply src/libraries/System.IO.Pipelines/src/PACKAGE.md:53]: cc @dotnet/area-system-io

**Pull request closed without merging**